scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3567,7 +3576,11 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
           val patterns = inParens(if (token.is[RightParen]) Nil else noSeq.patterns())
           makeTuple[Pat](patterns, () => Lit.Unit(), Pat.Tuple(_))
         case MacroQuote() =>
-          Pat.Macro(macroQuote())
+          val oldInPattern = QuotedPatternContext.inPattern
+          QuotedPatternContext.inPattern = true
+          val res = Pat.Macro(macroQuote())
+          QuotedPatternContext.inPattern = oldInPattern
+          res
```

Comment:
perhaps:
```
try {
  QuotedPatternContext.enter()
  Pat.Macro(macroQuote())
} finally {
  QuotedPatternContext.exit()
}
```

Reply:
Should I do this to the uses of enter/exit elsewhere too?

Reply:
ooh, tricky question :) and thank you for suggesting. my take is this:

if others (@tgodzik and/or @dos65) agree with my preference for this approach (poor man's "destructor") over something like
```
  enter()
  val res = ...
  exit()
  res
```
then it would be nice to make that change. i would, however, do it as a separate commit, to keep it clean.

Reply:
I will rebase, one sec.

Reply:
Okay, done.